### ISSUE TITLE
Cannot create Inter Company Purchase Invoice

### ISSUE THREAD

**Issue by rtdany10** [frappe/erpnext#30930]
### Information about bug

1. [Create an internal supplier and customer](https://docs.erpnext.com/docs/v13/user/manual/en/accounts/inter-company-invoices)
2. Create a Sales Invoice against the internal customer with `Update Stock` ticked
3. Try creating an inter company purchase invoice via the `Create` button

### Module

accounts

### Version

ERPNext: v13.27.0 (HEAD)
Frappe Framework: v13.26.1 (HEAD)

### Installation method

FrappeCloud

### Relevant log output / Stack trace / Full Error Message.

```shell
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 69, in application
    response = frappe.api.handle()
  File "apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 37, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 75, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1447, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/model/mapper.py", line 41, in make_mapped_doc
    return method(source_name)
  File "apps/erpnext/erpnext/accounts/doctype/sales_invoice/sales_invoice.py", line 2145, in make_inter_company_purchase_invoice
    return make_inter_company_transaction("Sales Invoice", source_name, target_doc)
  File "apps/erpnext/erpnext/accounts/doctype/sales_invoice/sales_invoice.py", line 2262, in make_inter_company_transaction
    set_missing_values,
  File "apps/frappe/frappe/model/mapper.py", line 149, in get_mapped_doc
    postprocess(source_doc, target_doc)
  File "apps/erpnext/erpnext/accounts/doctype/sales_invoice/sales_invoice.py", line 2165, in set_missing_values
    target.run_method("set_missing_values")
  File "apps/frappe/frappe/model/document.py", line 941, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1260, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1242, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 938, in fn
    return method_object(*args, **kwargs)
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 162, in set_missing_values
    super(PurchaseInvoice, self).set_missing_values(for_validate)
  File "apps/erpnext/erpnext/controllers/buying_controller.py", line 94, in set_missing_values
    self.set_missing_item_details(for_validate)
  File "apps/erpnext/erpnext/controllers/accounts_controller.py", line 543, in set_missing_item_details
    self.set_expense_account(for_validate)
  File "apps/erpnext/erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py", line 314, in set_expense_account
    item.expense_account = warehouse_account[item.warehouse]["account"]
KeyError: None
```

**Comment by ankush** [comment]
This feels more related to WH -> account mapping. Can you share anonymized values of `warehouse_account`?

**Comment by rtdany10** [comment]
> This feels more related to WH -> account mapping. Can you share anonymized values of `warehouse_account`?

I'm not able to replicate this on my dev server. FC provides me no way to print out that data either.
But according to [this code](https://github.com/frappe/erpnext/blob/v13.27.0/erpnext/stock/__init__.py#L78-L79), if no account is found, it should set default one from company master. So, the only thing I suspect of being `None` is `item.warehouse`.

Therefore, I tried setting **Item Defaults** in Item master for all the companies with their respective warehouses and I was able to generate the Purchase Invoice.

Not sure if I should close the issue or not.
@ankush
